File: implementations/go/capability/capability_test.go
package capability

import "testing"

func TestCapabilityEventTypesIncludesAllEntries(t *testing.T) {
	expected := []string{
		"capability.registered",
		"capability.updated",
		"capability.deprecated",
		"capability.composed",
		"capability.validated",
	}
	for _, typ := range expected {
		if !EventTypes[typ] {
			t.Errorf("missing: %s", typ)
		}
	}
	if len(EventTypes) != len(expected) {
		t.Errorf("size mismatch: got %d, want %d", len(EventTypes), len(expected))
	}
}

func TestIsEventTypePositives(t *testing.T) {
	tests := []string{
		"capability.registered",
		"capability.updated",
		"capability.deprecated",
		"capability.composed",
		"capability.validated",
	}
	for _, typ := range tests {
		if !IsEventType(typ) {
			t.Errorf("expected true for %s", typ)
		}
	}
}

func TestIsEventTypeNegatives(t *testing.T) {
	tests := []string{
		"capabilities.requested",
		"task.submitted",
		"command.requested",
		"",
	}
	for _, typ := range tests {
		if IsEventType(typ) {
			t.Errorf("expected false for %s", typ)
		}
	}
}

File: implementations/typescript/packages/capability/test/capability.test.js
import assert from "node:assert/strict";
import test from "node:test";
import { CAPABILITY_EVENT_TYPES, isCapabilityEventType } from "../src/index.js";

test("capability event types includes all 5 registry entries", () => {
  const expected = [
    "capability.registered",
    "capability.updated",
    "capability.deprecated",
    "capability.composed",
    "capability.validated"
  ];
  for (const type of expected) {
    assert.equal(CAPABILITY_EVENT_TYPES.has(type), true, `missing: ${type}`);
  }
  assert.equal(CAPABILITY_EVENT_TYPES.size, expected.length);
});

test("isCapabilityEventType positives", () => {
  assert.equal(isCapabilityEventType("capability.registered"), true);
  assert.equal(isCapabilityEventType("capability.updated"), true);
  assert.equal(isCapabilityEventType("capability.deprecated"), true);
  assert.equal(isCapabilityEventType("capability.composed"), true);
  assert.equal(isCapabilityEventType("capability.validated"), true);
});

test("isCapabilityEventType negatives", () => {
  assert.equal(isCapabilityEventType("capabilities.requested"), false);
  assert.equal(isCapabilityEventType("task.submitted"), false);
  assert.equal(isCapabilityEventType("command.requested"), false);
  assert.equal(isCapabilityEventType(""), false);
});
